stratify/_bounded_vinterp.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import absolute_import
+
+import numpy as np
+
+from ._conservative import conservative_interpolation
+
+
+def interpolate(z_target, z_src, fz_src, axis=-1):
+    """
+    Interface for optimised 1d interpolation across multiple dimensions.
+
+    This function provides the ability to perform 1d interpolation on datasets
+    with more than one dimension. For instance, this function can be used to
+    interpolate a set of vertical levels, even if the interpolation coordinate
+    depends upon other dimensions.
+
+    A good use case might be when wanting to interpolate at a specific height
+    for height data which also depends on x and y - e.g. extract 1000hPa level
+    from a 3d dataset and associated pressure field. In the case of this
+    example, pressure would be the `z` coordinate, and the dataset
+    (e.g. geopotential height / temperature etc.) would be `f(z)`.
+
+    Parameters
+    ----------
+    z_target: :class:`np.ndarray`
+        Target coordinate.
+        This coordinate defines the levels to interpolate the source data
+        ``fz_src`` to.  ``z_target`` must have the same dimensionality as the
+        source coordinate ``z_src``, and the shape of ``z_target`` must match
+        the shape of ``z_src``, although the axis of interpolation may differ
+        in dimension size.
+    z_src: :class:`np.ndarray`
+        Source coordinate.
+        This coordinate defines the levels that the source data ``fz_src`` is
+        interpolated from.
+    fz_src: :class:`np.ndarray`
+        The source data; the phenomenon data values to be interpolated from
+        ``z_src`` to ``z_target``.
+        The data array must be at least ``z_src.ndim``, and its trailing
+        dimensions (i.e. those on its right hand side) must be exactly
+        the same as the shape of ``z_src``.
+    axis: int (default -1)
+        The ``fz_src`` axis to perform the interpolation over.
+
+    Returns
+    -------
+    : :class:`np.ndarray`
+        fz_src interpolated from z_src to z_target.
+
+    Note
+    ----
+    Support for 1D z_target and corresponding ND z_src will be provided in
+    future as driven by user requirement.
+
+    """
+    if z_src.ndim != z_target.ndim:
+        msg = ('Expecting source and target levels dimensionality to be '
+               'identical.  {} != {}.')
+        raise ValueError(msg.format(z_src.ndim, z_target.ndim))
```

Comment:
Would this (and following assertions) be more transparent as:
```
('Expecting dimensionality of source levels {} and target levels {} to be '
               'identical.')
```
It's just moving the value of the "thing" to where the definition of the "thing" is.